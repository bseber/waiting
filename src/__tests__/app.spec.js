import React from "react";
import { renderShallow, renderIntoDocument } from "../../test/testHelpers";
import ReactDOM from "react-dom";
import ReactTestUtils, { Simulate } from "react-dom/test-utils";
import App from "../app";
import Waiting from "../waiting";

jest.useFakeTimers();

describe("App", () => {
    it("renders", () => {
        const tree = renderShallow(<App />);
        expect(tree).toMatchSnapshot();
    });

    it("renders fetching state", () => {
        const root = document.createElement("div");
        const tree = renderIntoDocument(<App />, root);
        const waiting = ReactTestUtils.findRenderedComponentWithType(tree, Waiting);

        expect(waiting.props.loading).toBe(false);

        const button = root.querySelector("button");
        Simulate.click(button);

        expect(waiting.props.loading).toBe(true);

        jest.runAllTimers();
        expect(waiting.props.loading).toBe(false);
    });
});
