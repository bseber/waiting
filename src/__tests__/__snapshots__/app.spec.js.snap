// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<UNDEFINED>
  <button
    onClick={[Function]}
  >
    start fetching with timeout (ms):
  </button>
  <input
    onChange={[Function]}
    type="number"
    value={2500}
  />
  <Waiting
    loading={true}
    render={[Function]}
  />
</UNDEFINED>
`;
